### PR TITLE
fix #6023, /v[x] with multi arg

### LINKED ISSUE #6023: Allow /v[x] to accept more than one value
the expected behaviour is to concatenate them honoring the endianness

example:

```
/v4 0x10203040 0x20304050

--> will search for 4030201050403020
```

### PR DESCRIPTION
**Pull request by sivaramaaa**
This PR aims to fix this issue [Allow /v[x] to accept more than one value](https://github.com/radare/radare2/issues/6023)

### CODE REVIEW

**Comment by radare** [comment]
I think we should reuse the num_list thing that was merged today for the ? Command

**Comment by sivaramaaa** [comment]
Ok , i will look into it !!

**Comment by Maijin** [comment]
@radare can we merge this so the test pass?

File: libr/core/cmd_search.c [hunk 1]
```diff
@@ -2754,34 +2754,53 @@ static int cmd_search(void *data, const char *input) {
...
+		int len = r_list_length (nums);
 		switch (input[1]) {
 		case '8':
 			if (input[param_offset]) {
-				n64 = r_num_math (core->num, input + param_offset);
-				ut8 buf[sizeof (ut64)];
-				r_write_le64 (buf, n64);
+				ut8 buf[sizeof (ut64) * len];
+				ut8 *ptr = buf;
+				for (i = 0; i < len; i++)
```
Comment (radare) [inline libr/core/cmd_search.c]: open brace at right

File: libr/core/cmd_search.c [hunk 2]
```diff
@@ -2754,34 +2754,53 @@ static int cmd_search(void *data, const char *input) {
...
 			r_config_get_i (core->config, "search.distance"));
+		RList *nums = r_num_str_split_list (input + 3);
+		int len = r_list_length (nums);
 		switch (input[1]) {
 		case '8':
 			if (input[param_offset]) {
-				n64 = r_num_math (core->num, input + param_offset);
-				ut8 buf[sizeof (ut64)];
-				r_write_le64 (buf, n64);
+				ut8 buf[sizeof (ut64) * len];
```
Comment (radare) [inline libr/core/cmd_search.c]: use the heap. variable sized local variables is the root of all evil

File: libr/core/cmd_search.c [hunk 3]
```diff
@@ -2754,34 +2754,53 @@ static int cmd_search(void *data, const char *input) {
...
 			} else {
 				eprintf ("Usage: /v8 value\n");
 			}
 			break;
 		case '1':
 			if (input[param_offset]) {
-				n8 = (ut8) r_num_math (core->num, input + param_offset);
+				ut8 buf[sizeof (ut8) * len];
+				ut8 *ptr = buf;
+				for (i = 0; i < len; i++)
```
Comment (radare) [inline libr/core/cmd_search.c]: open brace at right

File: libr/core/cmd_search.c [hunk 4]
```diff
@@ -2754,34 +2754,53 @@ static int cmd_search(void *data, const char *input) {
...
 			}
 			break;
 		case '2':
 			if (input[param_offset]) {
-				n16 = (ut16) r_num_math (core->num, input + param_offset);
-				ut8 buf[sizeof (ut16)];
-				r_write_le16 (buf, n16);
+				ut8 buf[sizeof (ut16) * len];
+				ut8 *ptr = buf;
+				for (i = 0; i < len; i++)
```
Comment (radare) [inline libr/core/cmd_search.c]: open brace at right

File: libr/core/cmd_search.c [hunk 5]
```diff
@@ -2754,34 +2754,53 @@ static int cmd_search(void *data, const char *input) {
...
 			} else {
 				eprintf ("Usage: /v1 value\n");
 			}
 			break;
 		case '2':
 			if (input[param_offset]) {
-				n16 = (ut16) r_num_math (core->num, input + param_offset);
-				ut8 buf[sizeof (ut16)];
-				r_write_le16 (buf, n16);
+				ut8 buf[sizeof (ut16) * len];
```
Comment (radare) [inline libr/core/cmd_search.c]: xxx

**Comment by Maijin** [comment]
updated

File: libr/core/cmd_search.c [hunk 6]
```diff
@@ -2321,9 +2321,45 @@ void _CbInRangeSearchV(RCore *core, ut64 from, ut64 to, int vsize, bool asterisk
...
 }
 
+static ut8 *v_writebuf(RCore *core, RList *nums, int len, char ch, int bsize) {
+	ut8 *buf, *ptr;
+	ut64 n64;
+	ut32 n32;
+	ut16 n16;
+	ut8 n8;
+	int i = 0;
+	buf = malloc (bsize);
```
Comment (radare) [inline libr/core/cmd_search.c]: Check for null

Comment (radare) [reply libr/core/cmd_search.c]: Char* in the same line

Comment (radare) [reply libr/core/cmd_search.c]: Ensure that ptr+delta+size <bsize

File: libr/core/cmd_search.c [hunk 7]
```diff
@@ -2321,9 +2321,45 @@ void _CbInRangeSearchV(RCore *core, ut64 from, ut64 to, int vsize, bool asterisk
...
+			n32 =  r_num_math (core->num, r_list_pop_head (nums));
+			r_write_le32 (ptr, n32);
+			ptr = (ut8 *) ptr + sizeof (ut32);
+			break;
+		case '8':
+			n64 = r_num_math (core->num, r_list_pop_head (nums));
+			r_write_le64 (ptr, n64);
+			ptr = (ut8 *) ptr + sizeof (ut64);
+			break;	
+		}
```
Comment (radare) [inline libr/core/cmd_search.c]: Default:

**Pull request merged**